ros_packages/motors/motors/motor_control.py

```diff
@@ -148,6 +152,36 @@ def apply_joint_trajectory(
             self.get_logger().error(f"error while applying joint-trajectory: {str(e)}")
         return response
 
+    def apply_pose(
+        self, request: ApplyPose.Request, response: ApplyPose.Response
+    ) -> ApplyPose.Response:
+        self.get_logger().info(f"Pose ID: {request.pose_id}")
+        successful, motor_positions = pose_client.get_motor_positions_of_pose(
+            request.pose_id
+        )
+        if not successful:
+            response.successful = False
+            return response
+
+        jt = JointTrajectory()
+        jt.joint_names = []
+
+        for motor_position in motor_positions["motorPositions"]:
+            motor_name = motor_position["motorName"]
+            position = motor_position["position"]
+
+            jt.joint_names.append(motor_name)
+            point = JointTrajectoryPoint()
+            point.positions.append(position)
+            jt.points.append(point)
+
+        req = ApplyJointTrajectory.Request()
+        req.joint_trajectory = jt
+        resp = ApplyJointTrajectory.Response()
+        resp.successful = resp.successful
+
+        return response
```

Comment:
Are poses actually getting applied? To me it looks like they don't, since there seems to be no interaction with the motor-objects.

I think we should do the following:

```
        successful, motor_positions = pose_client.get_motor_positions_of_pose(
            request.pose_id
        )
        if not successful:
            response.successful = False
            return response

        jt = JointTrajectory()
        jt.joint_names = []

        for motor_position in motor_positions["motorPositions"]:
            motor_name = motor_position["motorName"]
            position = motor_position["position"]

            jt.joint_names.append(motor_name)
            point = JointTrajectoryPoint()
            point.positions.append(position)
            jt.points.append(point)

        req = ApplyJointTrajectory.Request()
        req.joint_trajectory = jt
```
this code-block should be moved to the `APPLY_POSE_FUNCTION` defined in the pib-blockly-repo.
The `req` object that is generated in the code-block above, should then be used to call an `apply_joint_trajectory`-service in `APPLY_POSE_FUNCTION` (i.e.: replace the `apply_pose_service` with an `apply_joint_trajectory_service` and then pass the `req`-object to it). Since you are using the `pose_client` in this code-block, you also need to import the `pose_client` in the `moveToPose`-generator in pib-blockly (you can look at how the other imports are handled in pib-blockly for reference). Lastly, when you perform the `if not successful:` check in the code-block above, instead of setting the `response.successful`-value to `False`, you should instead log an error and return from the function prematurely.

Reply:
Is this still relevant?